packages/core/src/Stores/ui-store.js

```diff
@@ -397,6 +398,8 @@ export default class UIStore extends BaseStore {
             toggleLanguageSettingsModal: action.bound,
             toggleUnsupportedContractModal: action.bound,
             toggleUpdateEmailModal: action.bound,
+            prevent_render: observable,
+            setPreventRender: action.bound,
```

Comment:
Please revisit the logic to handle the scenario if there's a re-render happening in the component rather using a flag to prevent a render